src/server/pfs/server/url.go

```diff
@@ -95,10 +105,15 @@ func putFileURLRecursive(ctx context.Context, taskService task.Service, uw *file
 		if err != nil {
 			return errors.Wrapf(err, "error parsing url %v", src)
 		}
-		objClient, err := obj.NewClientFromURLAndSecret(ctx, url, false)
+		bucket, err := openBucket(ctx, url)
 		if err != nil {
 			return err
 		}
+		defer func() {
+			if err := bucket.Close(); err != nil {
+				retErr = multierror.Append(retErr, errors.Wrapf(err, "error closing bucket %s", url.Bucket))
```

Comment:
This needs to be operating on the error returned by the function being run in the error group, not the outer function.